### PR TITLE
Temporary fix for subnetwork exports

### PR DESCRIPTION
**Pull request by nickrsan**

### CODE REVIEW

File: nodejs/matrix/node.js [hunk 1]
```diff
@@ -36,44 +36,51 @@ module.exports = function(item, subnet) {
...
-          u.roundBound(edge.flow[i][1])
+          u.roundBound(edge.flow[flow_at_timestep_idx][1]),
+          u.roundBound(edge.flow[flow_at_timestep_idx][1])
         ]);
       }
+
+      // Attach outbound boundary conditions
       for (e = 0; e < outbound.length; e++) {
         edge = outbound[e].properties;
-        if( edge.flow.length >= i ) continue;
```
Comment (nickrsan) [inline nodejs/matrix/node.js:67]: @msdogan and @josue-medellin just calling your attention to this. I'd made this fix to the calvin-network-tools code back in 2020 to make regional exports/subnets work, but it involves moving/removing some code that I don't full understand the implications of. Can you take a look at the changes here and see if you think they're OK and that regional/subnet exports are still safe and the results are OK, or if we should undo them?

File: nodejs/matrix/node.js [hunk 2]
```diff
@@ -36,44 +36,51 @@ module.exports = function(item, subnet) {
...
-      steps.push(flow[i][0]);
+      // attach the current timestep to the output
+      steps.push(flow[flow_at_timestep_idx][0]);
 
       // Add Inflows from edge links
       // debugger;
       for (e = 0; e < inbound.length; e++) {
         edge = inbound[e].properties;
-        if( edge.flow.length >= i ) continue;
+        //if( edge.flow.length >= flow_at_timestep_idx ) continue; // commented out as part of issue #1 in UCM repo
```
Comment (nickrsan) [inline nodejs/matrix/node.js:56]: Another spot where we comment out a continue that skips edges.

**Comment by msdogan** [comment]
@nickrsan I exported a sub network for `sacramento-river\southeast` region. Below is the code I used to export. Although the code create sub networks work, when I run through CALVIN, it does not work. `cnf` command does not properly create boundary conditions. When we create sub networks, the idea was to put basecase CALVIN outputs as boundary conditions on links that were cut (only on boundary links). 

```
cnf matrix --data=/Users/msdogan/Documents/github/calvin-network-data/data  --verbose --format=csv --start=2002-10 --stop=2003-01 --ts=. --fs=, --to=network_one_year --outnodes=nodes --max-ub=1000000000 --debug=All --regions= sacramento-river/southeast
```

Error message from CALVIN:
```
no incoming link for C80.2002-10-31
```

**Comment by nickrsan** [comment]
OK, yeah - it sure sounds like we should revert this change then if it's exporting networks that can't be run through CALVIN.

I think we're back to a question we had in 2020 then too - how important are the subnetworks for continued use of the tool? I could take another crack at this code, but it was somewhat impenetrable previously.

**Comment by nickrsan** [comment]
Also, I mean to mention @josue-medellin here - Josue, it's worth you taking a look at the rest of this thread. Note that the beginning is from 2020, but the discussion between Mustafa and me is from this week.

**Pull request merged**